src/hooks/useDittoComponent.ts

```diff
@@ -0,0 +1,67 @@
+import { useContext } from "react";
+import { DittoContext } from "../lib/context";
+import { nullError } from "../lib/utils";
+
+interface useDittoComponentProps {
+  componentId: string;
+  alwaysReturnString: boolean;
+}
+
+export const useDittoComponent = (
+  props: useDittoComponentProps
+):
+  | string
+  | {
+      name: string;
+      text: string;
+    } => {
```

Comment:
can we clean this up into it's own interface

Reply:
cb9ee59